peerinst/views/collection.py

```diff
@@ -278,6 +280,20 @@ def get_context_data(self, **kwargs):
         return context
 
 
+@login_required
+@user_passes_test(student_check, login_url="/access_denied_and_logout/")
+def collection_statistics(request):
+    collection = get_object_or_404(Collection, pk=request.POST.get("pk"))
```

Comment:
How is this going to work in a view that returns a JSON response?  The 404 error will be an HTMLResponse